core/templates/pages/contributor-dashboard-page/contributor-badges/contributor-badges.component.ts

```diff
@@ -103,11 +103,16 @@ export class ContributorBadgesComponent {
     if (userContributionRights === null) {
       throw new Error('Cannot fetch user contribution rights.');
     }
-    userContributionRights.can_review_translation_for_language_codes.map(
-      (languageCode) => {
-        const languageDescription = this.languageUtilService
-          .getAudioLanguageDescription(
-            languageCode);
+
+    const allContributionStats = await this
+      .contributionAndReviewStatsService.fetchAllStats(username);
+
+    if (allContributionStats.translation_review_stats.length > 0) {
+      await allContributionStats.translation_review_stats.map((stat) => {
```

Comment:
Is `await` needed here? Isn't this just a synchronous transformation? Also using `.map()` here feels a bit odd here since the the value of the new Array is not being used. I think just call `forEach` is more idiomatic.

Reply:
Done